src/holder.rs

```diff
@@ -2,6 +2,52 @@ use crate::sync::atomic::AtomicPtr;
 use crate::{Domain, HazPtrObject, HazPtrRecord};
 use std::sync::atomic::Ordering;
 
+pub struct HazardPointerArray<'domain, F, const N: usize> {
+    domain: &'domain Domain<F>,
+    records: Option<[&'domain HazPtrRecord; N]>,
```

Comment:
I feel like I'm missing something. Why is this `Option`? I think it's because of `Drop`, but given that this is holding `[&..]`, it should be `Copy` anyway, so there's no need to `take()`?

Reply:
You're right, I didn't realize that array was Copy